src/platform/utilities/environment/index.js

```diff
@@ -1,43 +1,85 @@
+
+// eslint-disable-next-line no-undef
+const BUILDTYPE = __BUILDTYPE__;
+
+const {
+  WEB_PORT: RESERVED_E2E_PORT = 3333,
+  API_PORT: E2E_API_PORT = 3000,
+} = process.env;
+
+const ENVIRONMENT_CONFIGURATIONS = {
+  [ENVIRONMENTS.VAGOVPROD]: {
+    BUILDTYPE: ENVIRONMENTS.VAGOVPROD,
     BASE_URL: 'https://www.va.gov',
+    API_URL: 'https://api.va.gov',
   },
-  vagovstaging: {
-    API_URL: 'https://staging-api.va.gov',
+
+  [ENVIRONMENTS.VAGOVSTAGING]: {
+    BUILDTYPE: ENVIRONMENTS.VAGOVSTAGING,
     BASE_URL: 'https://staging.va.gov',
+    API_URL: 'https://staging-api.va.gov',
   },
-  vagovdev: {
-    API_URL: 'https://dev-api.va.gov',
+
+  [ENVIRONMENTS.VAGOVDEV]: {
+    BUILDTYPE: ENVIRONMENTS.VAGOVDEV,
     BASE_URL: 'https://dev.va.gov',
+    API_URL: 'https://dev-api.va.gov',
   },
-  localhost: {
-    API_URL: `http://${location.hostname}:3000`,
+
+  [ENVIRONMENTS.LOCALHOST]: {
+    BUILDTYPE: ENVIRONMENTS.LOCALHOST,
     BASE_URL: `http://${location.hostname}:3001`,
-  },
-  e2e: {
-    API_URL: `http://localhost:${process.env.API_PORT || 3000}`,
-    BASE_URL: `http://localhost:${process.env.WEB_PORT || 3333}`,
+    API_URL: `http://${location.hostname}:3000`,
   },
 };
 
-function getEnvironment() {
-  let platform;
+const environment = ENVIRONMENT_CONFIGURATIONS[BUILDTYPE];
+
+if (location.port === RESERVED_E2E_PORT) {
+  // E2E tests are an edge case - they test a certain build-type,
+  // but execute under the localhost hostname.
 
-  if (location.port === '3001') {
-    platform = 'localhost';
-  } else if (location.port === `${process.env.WEB_PORT || 3333}`) {
-    platform = 'e2e';
-  } else {
-    platform = __BUILDTYPE__;
-  }
+  const e2eConfig = {
+    API_URL: `http://localhost:${E2E_API_PORT}`,
+    BASE_URL: `http://localhost:${RESERVED_E2E_PORT}`,
+  };
 
-  return _Environments[platform];
+  Object.assign(environment, e2eConfig);
 }
 
-const environment = getEnvironment();
-module.exports = environment;
+export default {
+  /** The name of the environment under which the site is currently executing. */
+  BUILDTYPE: environment.BUILDTYPE,
+
+  /** The address of the FE website configured for this environment. */
+  BASE_URL: environment.BASE_URL,
+
+  /** The address of the API configured for this environment. */
+  API_URL: environment.API_URL,
+
+  /** Determines whether the current environment is a production environment. */
+  isProduction() {
+    return this.BUILDTYPE === ENVIRONMENTS.VAGOVPROD;
+  },
+
+  /** Determines whether the current environment is a staging environment. */
+  isStaging() {
+    return this.BUILDTYPE === ENVIRONMENTS.VAGOVSTAGING;
+  },
+
+  /** Determines whether the current environment is a dev environment. */
+  isDev() {
+    return this.BUILDTYPE === ENVIRONMENTS.VAGOVDEV;
+  },
+
+  /** Determines whether the current environment is a local environment. */
+  isLocalhost() {
+    return this.BUILDTYPE === ENVIRONMENTS.LOCALHOST;
+  },
+};
```

Comment:
I'm hoping that this helper functions will be able to serve as feature flags without needing to check the actual build-type, which have names that aren't especially meaningful at first.